src/parser/reader.rs

```diff
@@ -29,7 +29,9 @@ impl<'l> Reader<'l> {
         if !block(self) {
             return None;
         }
-        let content = &self.content[start..self.offset].trim();
+        let start = self.content.char_indices().nth(start).unwrap().0;
+        let offset = self.content.char_indices().nth(self.offset - 1).unwrap().0;
+        let content = &self.content[start..offset + 1].trim();
```

Comment:
The inclusive range, `..=`, would work here instead of `+ 1`, right? Let’s switch.